dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -1487,4 +1487,13 @@ def get_hidden(script)
 
     assert_equal 1, SectionHiddenStage.where(section_id: section.id).length
   end
+
+  test "should redirect when script has a redirect_to property" do
+    script = create :script
+    new_script = create :script
+    create(:script_level, script: new_script)
+
+    get :show, params: {script_id: script.name, stage_position: '1', id: '1'}
+    assert_redirected_to "/s/#{new_script.name}/stage/1/puzzle/1"
```

Comment:
By redirecting from `script/stage/1/puzzle/1` to `new_script/stage/1/puzzle/1` this test might accidentally imply that the script position is preserved in the redirect.  Might be worth modifying this test or adding a new test demonstrating that we always redirect back to the beginning of the new script.